### PR TITLE
lib60870: init at 2.3.2

### PR DESCRIPTION
**Pull request by stv0g**
## Description of changes

This PR packages [lib60870](https://lib60870.com/) an C implementation of the IEC 60870-5-101/104 protocol used in power system automation.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/lib60870/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ cmake
+, lib
+, stdenv
+, fetchzip
+}:
+stdenv.mkDerivation {
+  pname = "lib60870";
+  version = "2.3.1";
```
Comment (jopejoe1) [inline pkgs/by-name/li/lib60870/package.nix:8]: Newest version is 2.3.2 from https://github.com/mz-automation/lib60870/releases

File: pkgs/by-name/li/lib60870/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  separateDebugInfo = true;
+
+  nativeBuildInputs = [ cmake ];
+
+  preConfigure = "cd lib60870-C";
+
+  meta = with lib; {
+    description = "Implementation of the IEC 60870-5-101/104 protocol";
+    homepage = "https://libiec61850.com/";
+    license = licenses.gpl3;
```
Comment (jopejoe1) [inline pkgs/by-name/li/lib60870/package.nix:24]: gpl3 is deprecated on favor of gpl3Only and gpl3Plus
```suggestion
    license = licenses.gpl3Only;
```

**Review by jopejoe1 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/lib60870/package.nix (hunk 2 above)
Comment (stv0g) [reply pkgs/by-name/li/lib60870/package.nix]: Thanks. Addressed in e1370a466ca9174fe8ba5f699321d4db48dfefb4

File: pkgs/by-name/li/lib60870/package.nix (hunk 1 above)
Comment (stv0g) [reply pkgs/by-name/li/lib60870/package.nix]: Thanks. Addressed in e1370a466ca9174fe8ba5f699321d4db48dfefb4

File: pkgs/by-name/li/lib60870/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ cmake
+, lib
+, stdenv
+, fetchFromGitHub
+}:
+stdenv.mkDerivation {
```
Comment (jopejoe1) [inline pkgs/by-name/li/lib60870/package.nix:6]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/li/lib60870/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "lib60870";
+  version = "2.3.2";
+
+  src = fetchFromGitHub {
+    owner = "mz-automation";
+    repo = "lib60870";
+    rev = "53a6b3c1cf3023e51cf81763b1ccf048edcd1c64";
```
Comment (jopejoe1) [inline pkgs/by-name/li/lib60870/package.nix:13]:
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/by-name/li/lib60870/package.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  preConfigure = "cd lib60870-C";
+
+  meta = with lib; {
+    description = "Implementation of the IEC 60870-5-101/104 protocol";
+    homepage = "https://libiec61850.com/";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ stv0g ];
+    platforms = [ "x86_64-linux" ];
+  };
+}
```
Comment (jopejoe1) [inline pkgs/by-name/li/lib60870/package.nix:30]:
```suggestion
})
```

**Review by jopejoe1 — CHANGES_REQUESTED** [review]
Use finalAttrs to set rev to current version, and maybe also add a update script:

```nix
  passthru.updateScript = gitUpdater {
    rev-prefix = "v";
  };
```

File: pkgs/by-name/li/lib60870/package.nix (hunk 5 above)
Comment (stv0g) [reply pkgs/by-name/li/lib60870/package.nix]: Thanks. Addressed in 1a4591f71467ef88dedd2a7f7ba246abd185f1dd

File: pkgs/by-name/li/lib60870/package.nix (hunk 4 above)
Comment (stv0g) [reply pkgs/by-name/li/lib60870/package.nix]: Thanks. Addressed in 1a4591f71467ef88dedd2a7f7ba246abd185f1dd

File: pkgs/by-name/li/lib60870/package.nix (hunk 3 above)
Comment (stv0g) [reply pkgs/by-name/li/lib60870/package.nix]: Thanks. Addressed in 1a4591f71467ef88dedd2a7f7ba246abd185f1dd

**Comment by stv0g** [comment]
@jopejoe1 Do you maybe also want to review https://github.com/NixOS/nixpkgs/pull/298584?

Its a related library from MZ-Automation.

**Review by jopejoe1 — APPROVED** [review]

**Pull request merged**